packages/demand-bloc-trip-create/src/TripCreateForm.ts

```diff
@@ -1,49 +1,93 @@
+  TripCreateModuleOptions,
+} from './types'
 import { Form } from './Form'
+import { isDateTimeInFuture } from './tripCreateValidators'
+import { defaultTripCreateOptions } from './constants'
 
-const defaultOptions = {
-  minLengthToSearch: 3,
-  autocompleteDebounceTime: 500,
-  autocompleteLocationRadius: 1100000,
-}
-
-export class TripCreateForm {
-  static create(locationService: Locations, options?: TripCreateBlocOptions) {
-    const mergedOptions = Object.assign({}, options, defaultOptions)
+export const dropoffSelectedKey = `${TripCreateFormFields.DROPOFF}.selectedAddress`
+export const pickupSelectedKey = `${TripCreateFormFields.PICKUP}.selectedAddress`
 
-    const formSchema = {
-      [TripCreateFormFields.PICKUP]: {
-        type: TripCreateFieldTypes.AUTOCOMPLETE,
-        locationService,
-        options: mergedOptions,
-      },
-      [TripCreateFormFields.DROPOFF]: {
-        locationService,
-        type: TripCreateFieldTypes.AUTOCOMPLETE,
-        options: mergedOptions,
-      },
-      [TripCreateFormFields.TRAIN_NUMBER]: {
-        type: TripCreateFieldTypes.GENERIC,
-        // TODO: fix type to not have this fields if type is Generic
-        locationService,
-        options: mergedOptions,
-      },
-    }
+export const getValidationSchema = (options: TripCreateModuleOptions) => {
+  const locationValidationSchema = yup
+    .object()
+    .shape({
+      placeId: yup.string().required(),
+      address: yup.object().shape({
+        displayAddress: yup.string().required(),
+      }),
+      timeZone: yup.string().required(),
+    })
+    .required()
 
-    const validationSchema = yup.object({
+  return yup
+    .object({
       [TripCreateFormFields.PICKUP]: yup
         .string()
-        .min(mergedOptions.minLengthToSearch)
+        .min(options.minLengthToSearch)
         .required(),
       [TripCreateFormFields.DROPOFF]: yup
         .string()
-        .min(mergedOptions.minLengthToSearch)
+        .min(options.minLengthToSearch)
         .required(),
-      [TripCreateFormFields.TRAIN_NUMBER]: yup.string(),
+      [dropoffSelectedKey]: locationValidationSchema,
+      [pickupSelectedKey]: locationValidationSchema,
+      [TripCreateFormFields.TRAIN_NUMBER]: yup.string().max(options.maxLengthForFlightAndTrainNumber),
+      [TripCreateFormFields.FLIGHT_NUMBER]: yup.string().max(options.maxLengthForFlightAndTrainNumber),
+      [TripCreateFormFields.PICKUP_DATE]: yup.mixed().when(TripCreateFormFields.IS_ASAP_BOOKING, {
+        is: false,
+        then: yup
+          .date()
+          .max(dayjs().add(1, 'year'))
+          .required(),
+      }),
+      [TripCreateFormFields.PICKUP_TIME]: yup.mixed().when(TripCreateFormFields.IS_ASAP_BOOKING, {
+        is: false,
+        then: yup.string().required(),
+      }),
+    })
+    .test('not-equal-addresses', 'not-equal-addresses', values => {
+      return values[pickupSelectedKey].placeId !== values[dropoffSelectedKey].placeId
     })
```

Comment:
Can be simplified a bit
`values => values[pickupSelectedKey].placeId !== values[dropoffSelectedKey].placeId)`